src/pudl_archiver/archivers/classes.py

```diff
@@ -245,7 +249,86 @@ def _check_missing_files(
             description="Check for files from previous version of archive that would be deleted by the new version",
             success=len(missing_files) == 0,
             notes=notes,
-            required_for_run_success=self.check_missing_files,
+            required_for_run_success=self.fail_on_missing_files,
+        )
+
+    def _check_file_size(
+        self,
+        baseline_datapackage: DataPackage | None,
+        new_datapackage: DataPackage,
+    ) -> validate.DatasetSpecificValidation:
+        """Check if any one file's size has changed by |>allowed_file_rel_diff|."""
+        baseline_resources = set()
+        too_changed_files = set()
+        notes = None
+
+        if baseline_datapackage is not None:
```

Comment:
non-blocking: it might be clearer that `baseline_datapackage is None` implies that this will always pass if you return early from that case:

```
if baseline_datapackage is None:
    return validate.DatasetSpecificValidation(...)
```

vs. populating this value only if baseline datapackage is not None. Then you can also avoid having to initialize `baseline_resources` to an empty value before assigning it an actual value.

I think it's more lines of code, because of the various params getting passed into `DatasetSpecificValidation`, but it's a little bit less Actual Logic.

Reply:
I've made a change halfway towards your suggestion, cleaning things up without fully duplicating the return call.